<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Ben Fair" />


<title>TCGA_sf3b1_reproduce_seiller</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">snaptron-analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">TCGA_sf3b1_reproduce_seiller</h1>
<h4 class="author">Ben Fair</h4>
<h4 class="date">6/24/2019</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-06-24
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>snaptron-analysis/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.4.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190618code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190618)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190618code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190618)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunsnaptronanalysistree6a283f40083c1e15d4b624a65de4414b11200daftargetblank6a283f4a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bfairkun/snaptron-analysis/tree/6a283f40083c1e15d4b624a65de4414b11200daf" target="_blank">6a283f4</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunsnaptronanalysistree6a283f40083c1e15d4b624a65de4414b11200daftargetblank6a283f4a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  analysis/TCGA_sf3b1_reproduce_seiller.Rmd
    Untracked:  docs/figure/

Unstaged changes:
    Modified:   _workflowr.yml
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(data.table)
library(dplyr)
library(ggplot2)
library(reshape2)
library(readxl)
library(ggrepel)
library(tidyverse)
library(gplots)
library(&quot;RColorBrewer&quot;)</code></pre>
<pre class="r"><code>Numers &lt;- &#39;../output/snakemake_out_sf3b1_TCGA/CountTableNumerators.gz&#39;
Denoms &lt;- &#39;../output/snakemake_out_sf3b1_TCGA/CountTableDenominators.gz&#39;

NumeratorCountTable &lt;- fread(paste0(&quot;gunzip -c &quot;, Numers), sep=&quot; &quot;, header=TRUE, data.table=FALSE)
DenominatorCountTable &lt;- fread(paste0(&quot;gunzip -c &quot;, Denoms), sep=&quot; &quot;, header=TRUE, data.table=FALSE)

PSI_Table &lt;- NumeratorCountTable[-1]/DenominatorCountTable[-1]
row.names(PSI_Table) &lt;- gsub(&quot;:clu_.+&quot;, &quot;&quot;, NumeratorCountTable$chrom)

#Replace NaNs from 0/0 operation
PSI_Table[is.na(PSI_Table)] &lt;- 0

# DifferentialJunctionsList &lt;- read.table(&#39;/Users/benfair/Downloads/Seiler_S3B1InBRCA_DifferentialJunctions.txt&#39;, header=F)
DifferentialJunctionsList &lt;- read.table(&#39;../data/other_snakemake_data/DarmanDiffJunctions.LiftedToHg38.bed&#39;, header=F)

# Make new id, for easier use in later filtering. Note that coordinates may need to add 1
DifferentialJunctionsList$junc_id &lt;- paste(DifferentialJunctionsList$V1, DifferentialJunctionsList$V2 + 1, DifferentialJunctionsList$V3 + 1, sep=&quot;:&quot;)

# PCA using PSI
PCResults &lt;- (PSI_Table) %&gt;%
  filter(rownames(.) %in% DifferentialJunctionsList$junc_id) %&gt;%
  t() %&gt;%
  prcomp(center=T, scale=T)
PCResults.df &lt;- as.data.frame(PCResults$x[,1:10]) %&gt;% rownames_to_column()

#Read in metadata for TCGA
TCGAMetadata &lt;- read.delim(gzfile(&#39;../output/snakemake_out_sf3b1_TCGA/SnaptronMetadata.TCGA.txt.gz&#39;), row.names=1, na.strings = &quot;NA&quot;, stringsAsFactors = T, header=T)

# truncated TCGA barcodes are easier to match and merge with Seiller et al metadata
TCGAMetadata$TruncatedTCGA &lt;- gsub(&quot;-[0-9]{2}[A-Z]-[0-9]{2}[A-Z]&quot;, &quot;&quot;, TCGAMetadata$gdc_cases.samples.portions.analytes.submitter_id)
TCGAMetadata$rail_id &lt;- row.names(TCGAMetadata)
TCGAMetadata$source &lt;- &quot;TCGA&quot;

MutationList &lt;- data.frame(read_excel(&quot;../data/other_snakemake_data/SeillerEtAl_MutationList.xlsx&quot;))
MutationList$TruncatedTCGA &lt;- gsub(&quot;-[0-9]{2}[A-Z]-[0-9]{2}[A-Z]-\\w{4}-\\w{2}$&quot;, &quot;&quot;, MutationList$Tumor_Sample_Barcode)

# Add labels to help distinguish SF3B1 hotspot mutations from other SF3B1 mutations in plot
# MutationList$Mutant_aa &lt;- paste(MutationList$Hugo_Symbol, MutationList$Protein_position, MutationList$Amino_acids, sep=&quot;:&quot;)
MutationList$Mutant_aa &lt;- paste(MutationList$Protein_position, MutationList$Amino_acids, sep=&quot;:&quot;)

MutationList$Mutant_aa[which(MutationList$Hugo_Symbol!=&quot;SF3B1&quot;)] &lt;- &quot;&quot;</code></pre>
<p>Ok so far I have read in the splice junction count tables created by leafcutter (a numerator and a denominator table, which when divided is a PSI-like metric of splicing). Furthermore, I have used TCGA barcodes to match the samples to their splicing factor mutation status as documented in a supplemental table from (Seiller et al 2018).</p>
<p>Now let’s plot SF3B1 mutation positions grouped by cancer sub-type. I expect most cancers to show a hotspot around amino acids 600-750, but different cancer types will have different mutation frequencies.</p>
<pre class="r"><code>#Plot hotspots
MutationList %&gt;%
  # filter(Cohort %in% c(&quot;LAML&quot;, &quot;SKCM&quot;, &quot;BRCA&quot;, &quot;BLCA&quot;, &quot;THYM&quot;, &quot;UCEC&quot;, &quot;UVM&quot;)) %&gt;%
  filter(Hugo_Symbol %in% c(&quot;SF3B1&quot;)) %&gt;%
  ggplot(aes(as.numeric(Protein_position))) +
  geom_histogram(aes(fill=Cohort_simplified),position=&quot;dodge&quot;, bins=50) +
  theme_bw()</code></pre>
<p><img src="figure/TCGA_sf3b1_reproduce_seiller.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  # geom_density(aes(color=Cohort_simplified))</code></pre>
<p>Ok good. Now let’s plot all the samples in the same PC space based on the differentially spliced introns identified in Darman et al. Same as to what is observed in Seiller et al, I expect this PC space will distinguish SF3B1 mutant from wildtype, and the 902 hotspot will not separate in this PC space as it is a functionally distinct mutation.</p>
<pre class="r"><code># MutationList$Cohort_simplified
Metadata &lt;- MutationList %&gt;%
  filter(Amino_acids != &quot;.&quot;) %&gt;%
  distinct(TruncatedTCGA, .keep_all=TRUE) %&gt;%
  right_join(TCGAMetadata, by = &quot;TruncatedTCGA&quot;, copy = FALSE)

ToPlot &lt;- PCResults.df %&gt;%
  mutate(rail_id=gsub(&quot;.project2.yangili1.snaptron.processed_to_bedfiles.TCGA.&quot;,&quot;&quot;, rowname)) %&gt;%
  left_join(Metadata, by=&quot;rail_id&quot;) %&gt;%
  mutate(CancerType=gsub(&quot;TCGA-&quot;,&quot;&quot;, gdc_cases.project.project_id))


# gdc_cases.project.project_id
ggplot(ToPlot, aes(x=PC1, y=PC2, color=CancerType, label=Mutant_aa)) +
  geom_text(alpha=1, size=2.2) +
  # geom_point(alpha=0.5) +
  # geom_text_repel(size=2.2) +
  theme_bw() +
  theme(legend.position=&quot;none&quot;)</code></pre>
<p><img src="figure/TCGA_sf3b1_reproduce_seiller.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok good. there it seems that the first PC does distinguish mut from wt, and later PCs seem to be separating tumor/tissue type.</p>
<p>It will also be nice to look at a heatmap of this feature space from which the PCA is based on… I will do this for only ~50 SF3B1 mutant samples (PC1&lt;-15) and 150 SF3B1 wildtype samples (PC1&gt;-15)… If I included all the samples it would be too dense a heatmap</p>
<pre class="r"><code>set.seed(0)
Cases&lt;-ToPlot %&gt;%
  filter(PC1 &lt;= -15) %&gt;%
  filter(Hugo_Symbol==&quot;SF3B1&quot; | is.na(Hugo_Symbol)) %&gt;%
  dplyr::pull(rowname)

Controls&lt;-ToPlot %&gt;%
  filter(PC1 &gt;= -15) %&gt;%
  filter(is.na(Hugo_Symbol)) %&gt;%
  sample_n(150) %&gt;%
  dplyr::pull(rowname)


Heatmap &lt;- PSI_Table %&gt;%
  filter(rownames(.) %in% DifferentialJunctionsList$junc_id) %&gt;%
  dplyr::select(c(Cases, Controls)) %&gt;% as.matrix() %&gt;%
  t() %&gt;% scale(scale=F) %&gt;% t() %&gt;% as.data.frame() %&gt;% drop_na() %&gt;% as.matrix()

#
# Heatmap &lt;- t(scale(t(as.matrix(PSI_Table[1:10, c(Cases, Controls)]))))
# Heatmap &lt;- scale(as.matrix(PSI_Table[1:50,c(Cases, Controls)]))

RNAExtractionDate &lt;- as.character(unclass(factor(plyr::mapvalues(colnames(Heatmap), from=ToPlot$rowname, to=ToPlot$CancerType, warn_missing = FALSE) )))
RNAExtractionDate &lt;- as.character(unclass(factor(plyr::mapvalues(colnames(Heatmap), from=ToPlot$rowname, to=ToPlot$Hugo_Symbol, warn_missing = FALSE) )))
ColLabel = labCol &lt;- plyr::mapvalues(colnames(Heatmap), from=ToPlot$rowname, to=ToPlot$Mutant_aa, warn_missing = FALSE)
# heatmap.2(Heatmap, col=brewer.pal(11,&quot;Spectral&quot;), dendrogram=&quot;column&quot;, trace=&quot;none&quot;, labRow=&quot;&quot;, labCol=ColLabel, ColSideColors=RNAExtractionDate, key.title=&quot;PSI-Zscore&quot;, ylab=&quot;574 splice junctions&quot;, xlab=&quot;200 samples&quot;)
heatmap.2(Heatmap, col=colorRampPalette(c(&quot;blue&quot;, &quot;white&quot;, &quot;red&quot;))(n = 299), dendrogram=&quot;column&quot;, trace=&quot;none&quot;, labRow=&quot;&quot;, labCol=ColLabel, ColSideColors=RNAExtractionDate, key.title=&quot;PSI-Zscore&quot;, ylab=&quot;574 splice junctions&quot;, xlab=&quot;200 samples&quot;)</code></pre>
<p><img src="figure/TCGA_sf3b1_reproduce_seiller.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Interestingly there are some samples where the genotype doesn’t match the phenotype. SF3B1. Could be incorrect genotype calls, or cases where the “hotspot” mutations are in hotspot region but not actually a phenotypic mutation.</p>
<p>What are the mutations that are in the hotspot region (amino acids 600-750) but don’t cluster with the mutant SF3B1 samples in PC1?</p>
<pre class="r"><code>#sf3b1 mut samples where PC1&lt;-5
ToPlot %&gt;%
  filter(Hugo_Symbol==&quot;SF3B1&quot; &amp; as.numeric(Protein_position) &gt;= 600 &amp; as.numeric(Protein_position) &lt;= 750) %&gt;%
  filter(PC1 &lt;= -5) %&gt;%
  dplyr::select(Mutant_aa)</code></pre>
<pre><code>   Mutant_aa
1    625:R/C
2    625:R/C
3    625:R/C
4    622:E/Q
5    625:R/C
6    700:K/E
7    666:K/E
8    700:K/E
9    666:K/T
10   633:A/V
11   700:K/E
12   625:R/C
13   700:K/E
14   740:G/E
15   662:H/R
16   625:R/H
17   625:R/H
18   625:R/H
19   625:R/C
20   625:R/C
21   700:K/E
22   625:R/H
23   666:K/T
24   625:R/H
25   700:K/E
26   700:K/E
27   625:R/H
28   625:R/H
29   626:N/Y
30   663:T/P
31   700:K/E
32   700:K/E
33   625:R/H
34   700:K/E
35   625:R/H
36   626:N/D
37   625:R/H
38   666:K/N
39   666:K/N
40   700:K/E
41   625:R/C
42   625:R/H
43   700:K/E
44   700:K/E
45   700:K/E
46   625:R/H
47   699:Q/E
48   700:K/E
49   625:R/H</code></pre>
<pre class="r"><code>#sf3b1 mut samples where PC1&gt;-5
ToPlot %&gt;%
  filter(Hugo_Symbol==&quot;SF3B1&quot; &amp; as.numeric(Protein_position) &gt;= 600 &amp; as.numeric(Protein_position) &lt;= 750) %&gt;%
  filter(PC1 &gt;= -5) %&gt;%
  dplyr::select(Mutant_aa)</code></pre>
<pre><code>  Mutant_aa
1   669:Q/H
2   736:R/S
3   700:K/E
4   697:E/*
5   693:G/D</code></pre>
<p>Ok it seems there are about 50 mutant samples that clearly cluster and are phenotypic for the same differentially spliced junctions identified in Darman et al. However there are 5 samples that have mutations between positions 600-750 that are not phenotypic. This is decent motivation for the premise of this project: namely that there is utility to targeted sequencing screening a patient’s RNA since clearly some “hotspot” mutations are not of significance; genotyping alone is not perfect at separating molecular separating phenotypes.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.14

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] RColorBrewer_1.1-2 gplots_3.0.1.1     forcats_0.4.0     
 [4] stringr_1.4.0      purrr_0.3.2        readr_1.3.1       
 [7] tidyr_0.8.3        tibble_2.1.3       tidyverse_1.2.1   
[10] ggrepel_0.8.1      readxl_1.3.1       reshape2_1.4.3    
[13] ggplot2_3.1.1      dplyr_0.8.1        data.table_1.12.2 

loaded via a namespace (and not attached):
 [1] gtools_3.8.1       tidyselect_0.2.5   xfun_0.7          
 [4] haven_2.1.0        lattice_0.20-38    colorspace_1.4-1  
 [7] generics_0.0.2     htmltools_0.3.6    yaml_2.2.0        
[10] rlang_0.3.4        pillar_1.4.1       glue_1.3.1        
[13] withr_2.1.2        modelr_0.1.4       plyr_1.8.4        
[16] munsell_0.5.0      gtable_0.3.0       workflowr_1.4.0   
[19] cellranger_1.1.0   rvest_0.3.4        caTools_1.17.1.2  
[22] evaluate_0.14      labeling_0.3       knitr_1.23        
[25] broom_0.5.2        Rcpp_1.0.1         KernSmooth_2.23-15
[28] scales_1.0.0       backports_1.1.4    gdata_2.18.0      
[31] jsonlite_1.6       fs_1.3.1           hms_0.4.2         
[34] digest_0.6.19      stringi_1.4.3      grid_3.5.1        
[37] rprojroot_1.3-2    bitops_1.0-6       cli_1.1.0         
[40] tools_3.5.1        magrittr_1.5       lazyeval_0.2.2    
[43] crayon_1.3.4       pkgconfig_2.0.2    xml2_1.2.0        
[46] lubridate_1.7.4    assertthat_0.2.1   rmarkdown_1.13    
[49] httr_1.4.0         rstudioapi_0.10    R6_2.4.0          
[52] nlme_3.1-140       git2r_0.25.2       compiler_3.5.1    </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
